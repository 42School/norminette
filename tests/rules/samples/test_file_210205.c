const char	*(*func)(void);

int	render_map(void *param)
{
	g->spritedist[i] = ((g->posx - sprite.x) * (g->posx - sprite.x) + 1);
}
