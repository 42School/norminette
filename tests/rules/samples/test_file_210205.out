[36mtest_file_210205.c[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 1":
		<CONST> <SPACE> <CHAR> <TAB> <MULT> <LPARENTHESIS> <MULT> <IDENTIFIER=func> <RPARENTHESIS> <LPARENTHESIS> <VOID> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_210205.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 2":
		<NEWLINE>
[36mtest_file_210205.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 3":
		<INT> <TAB> <IDENTIFIER=render_map> <LPARENTHESIS> <VOID> <SPACE> <MULT> <IDENTIFIER=param> <RPARENTHESIS> <NEWLINE>
[36mtest_file_210205.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 4":
		<LBRACE> <NEWLINE>
[36mtest_file_210205.c[0m - [32mIsAssignation[0m In "Function" from "GlobalScope" line 5":
		<TAB> <IDENTIFIER=g> <PTR> <IDENTIFIER=spritedist> <LBRACKET> <IDENTIFIER=i> <RBRACKET> <SPACE> <ASSIGN> <SPACE> <LPARENTHESIS> <LPARENTHESIS> <IDENTIFIER=g> <PTR> <IDENTIFIER=posx> <SPACE> <MINUS> <SPACE> <IDENTIFIER=sprite> <DOT> <IDENTIFIER=x> <RPARENTHESIS> <SPACE> <MULT> <SPACE> <LPARENTHESIS> <IDENTIFIER=g> <PTR> <IDENTIFIER=posx> <SPACE> <MINUS> <SPACE> <IDENTIFIER=sprite> <DOT> <IDENTIFIER=x> <RPARENTHESIS> <SPACE> <PLUS> <SPACE> <CONSTANT=1> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mtest_file_210205.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 6":
		<RBRACE> <NEWLINE>
test_file_210205.c: Error!
Error: INVALID_HEADER       (line:   1, col:   1):	Missing or invalid 42 header
